### ISSUE TITLE
Word motion grabs whitespace

### ISSUE THREAD

**Issue by joshbode** [mjl-/vijs#3]
The word motion `w` (and `e`, `E` and `W`) grabs adjoining whitespace, including newlines.

e.g. if the cursor is at the beginning of the word "foo":
```
yo
foo

bar
```
and the command `cw` is issued, the result is a reduction in the number of lines from 4 to 2:
```
yo
bar
```
with the cursor in front of the word "bar".

The expected result is:
```
yo


bar
```
with the cursor on the second line, and no change in the total number of lines.

This behaviour is consistent whether with `d`, `c` or `y` (which copies the empty lines)

**Comment by mjl-** [comment]
Thanks for this one too. "cw" makes the "w" motion work like "iw" (inner word), excluding trailing whitespace. Regular "w" would include the whitespace. At least that's how it works in my mind. (: 
There may be more combinations of "c" plus motion that need this change. Would be good to find docs somewhere that explain the mental model of the motions (including for visual select) and whether they have special interactions with certain commands.

Will be making a release with these changes now.

**Comment by joshbode** [comment]
Thanks @mjl- - I've just tested and this seems to work exactly like I expect!

In testing this, I've noticed one thing which may or may not be related, but is a difference to `vi`/`vim`. Namely, that `dd` will not remove the final line in a piece of text - it just leaves an empty line in place.
`dd` on any line other than the final line works correctly though and removes the line entirely.
I can open another issue, though :)
